From a2ad004ef95112e4311c02943ce40c2992597218 Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Tue, 28 Jun 2016 18:26:24 -0400
Subject: [PATCH 1820/2094] drm/amdgpu: Don't wait for fence when
 unmapping/freeing buffers

When unmapping and freeing buffers, user mode doesn't need to wait
for the completion of the page table update fence.

Change-Id: Idd948d1f3d7a7000b3845d16a81dbd2f8a9e8a8c
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 4ed616b..b1e1598 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -874,7 +874,7 @@ int amdgpu_amdkfd_gpuvm_free_memory_of_gpu(
 				entry);
 	}
 
-	unreserve_bo_and_vms(&ctx, true);
+	unreserve_bo_and_vms(&ctx, false);
 
 	/* Free the BO*/
 	amdgpu_bo_unref(&mem->data2.bo);
@@ -1210,7 +1210,7 @@ int amdgpu_amdkfd_gpuvm_unmap_memory_from_gpu(
 	}
 
 unreserve_out:
-	unreserve_bo_and_vms(&ctx, true);
+	unreserve_bo_and_vms(&ctx, false);
 out:
 	mutex_unlock(&mem->data2.lock);
 	return ret;
-- 
2.7.4

